packages/sprinkles/src/createSprinkles.ts

```diff
@@ -60,16 +60,35 @@ type SprinkleProps<Args extends ReadonlyArray<any>> = Args extends [
       SprinkleProps<R>
   : {};
 
+type SprinkleConditions<Args extends ReadonlyArray<any>> = Args extends [
+  infer L,
+  ...infer R
+]
+  ? L extends SprinklesProperties
+    ? [
+        keyof L['conditions'],
+        ...SprinkleConditions<[ChildSprinkleProps<L['styles']>]>,
+        ...SprinkleConditions<R>
+      ]
+    : []
+  : [];
+
 export type SprinklesFn<Args extends ReadonlyArray<SprinklesProperties>> = ((
   props: SprinkleProps<Args>,
-) => string) & { properties: Set<keyof SprinkleProps<Args>> };
+) => string) & {
+  properties: Set<keyof SprinkleProps<Args>>;
+  conditions: Set<SprinkleConditions<Args>>;
```

Comment:
Could you please add a type test for this? You can see this existing ones in `tests/sprinkles/sprinkles-type-tests.ts`.